Renamed IdentityRequestProvider to DigitalCredentialsProvider

marcoscaceres opened the pull request:
Closes #70

Implementation commitment:

- [ ] [WebKit](https://bugs.webkit.org/show_bug.cgi?id=277323)
- [ ] [Chromium](https://g-issues.chromium.org/issues/357094234)
- [ ] Gecko (link to issue)


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/WICG/digital-credentials/pull/71.html" title="Last updated on Aug 2, 2024, 5:19 PM UTC (515eed4)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/digital-credentials/71/6192ec4...515eed4.html" title="Last updated on Aug 2, 2024, 5:19 PM UTC (515eed4)">Diff</a>

timcappalli reviewed (approved):
makes sense to me

marcoscaceres commented on index.html:
```diff
@@ -169,7 +169,7 @@ <h2>
     <pre class="idl">
     partial dictionary CredentialRequestOptions {
-      sequence&lt;IdentityRequestProvider&gt; providers;
+      sequence&lt;IdentityRequestDetails&gt; providers;
```
I think we still want "providers" here, or maybe even something more specific (e.g., maybe even `identityProviders`). The "providers" is the specific key we route inside the browser to say "this request is about digital identities". 

For comparison, see `CredentialRequestOptions` has these members:

```WebIDL
partial dictionary CredentialRequestOptions {   FederatedCredentialRequestOptions federated; };
partial dictionary CredentialRequestOptions {   boolean password = false; };
```

The member's name basically indicates what UI the browser needs to show.

marcoscaceres replied:
Or, maybe even more generalized: `identities`.

samuelgoto replied:
I like `providers` better, but other options I think could be `requests` or `sources`?

marcoscaceres replied:
Let's keep providers for now... I'll file a bug that we might need to revise. We can get the Cred Man folks to also help us pick a better name if needed.

marcoscaceres replied:
Filed https://github.com/WICG/digital-identities/issues/74

tplooker replied:
I agree with Brian the type for the items in the "provider" array shouldn't have a divergent name, I'd prefer we keep it as IdentityCredentialProvider.

bc-pi replied:
IdentityCredentialProvider isn't a provider in the sense of the way provider is used elsewhere either. Its a request protocol thing.

samuelgoto commented on index.html:
```diff
@@ -182,15 +182,15 @@ <h3>
     </p>
     <h2>
-      The `IdentityRequestProvider` dictionary
+      The `IdentityRequestDetails` dictionary
```
Can we call this `DigitalIdentityRequest` as opposed to use the `Details` suffix? `Details` seems a bit off to me (although, surprisingly, there is a lot of "Details" in chrome, so maybe that's just me). Anyway, I think it is likely that whatever name you pick for this dictionary is likely to be temporary, because we need to form an opinion of what a `providers` is, so maybe we can just rename this when we figure out what to call `providers`?

[1] https://source.chromium.org/search?q=file:idl%20Details%20-file:out&sq=&ss=chromium

marcoscaceres replied:
yeah, that might be better indeed.

marcoscaceres commented:
Blocked on https://github.com/WICG/digital-identities/issues/74

msporny commented on index.html:
```diff
@@ -182,15 +182,15 @@ <h3>
     </p>
     <h2>
-      The `IdentityRequestProvider` dictionary
+      The `IdentityRequestDetails` dictionary
```
```suggestion
      The `PresentationRequest` dictionary
```

A request for a digital movie ticket is not a request for a digital identity... it's a request for the presentation of a digital credential.

Terminology that seems to have more consensus than "digital identity request" is the concept of a "presentation" (this is most likely  across mDL, mDoc, VCs, AnonCreds, and other emerging digital credential formats).  

Has this terminology been considered before?

https://w3c.github.io/vc-data-model/#dfn-presentation

marcoscaceres replied:
As the dictionary name is not exposed anywhere, it's only really helpful on the C++ side (and potentially via the browser's dev tools) - so generally what we want is something pretty specific here, hence the updated `DigitalCredentialsProvider` name.

msporny reviewed (changes requested):
We should consider the "presentation" terminology. That has been a hard won term of art over the last 7+ years and seems to be applicable to all digital credential ecosystems.

marcoscaceres commented:
Given we moved away from the whole "identity" thing, I renamed it to `DigitalCredentialsProvider` instead. That makes more sense in the context of `providers` being an array. 

I don't love "providers" tbh... but it's fine for now.

marcoscaceres commented:
Filed webkit bug https://bugs.webkit.org/show_bug.cgi?id=277323

timcappalli reviewed (approved):
Makes sense to me

timcappalli commented:
What if we called it `DigitalCredentialsPresentationRequest`? It is long but a bit better than providers IMO.

```
dictionary DigitalCredentialRequestOptions {
  sequence<DigitalCredentialsPresentationRequest> presentationRequests;
};
```

/cc @marcoscaceres @bc-pi

bc-pi commented:
> What if we called it `DigitalCredentialsPresentationRequest`? It is long but a bit better than providers IMO.

Agree it's better than providers. Don't know if it's *too* long.

TallTed commented:
I'd like to see consistency between singular and plural naming, and generally avoiding plurals, so here, `DigitalCredentialPresentationRequest`.

samuelgoto reviewed (approved):
LGTM

These are just dictionary names and are not web exposed. The names you picked are much better from what they currently are, consistent with the rest of the APIs.

marcoscaceres commented:
Ok, let's go with this for now. These are easy to change if we need to and, as stated, they are not web exposed things.
